improver_tests/utilities/test_save.py

```diff
@@ -235,6 +236,45 @@ def test_error_unknown_units(self):
             save_netcdf(no_units_cube, self.filepath)
 
 
+@pytest.fixture(name="bitshaving_cube")
+def bitshaving_cube_fixture():
+    """ Sets up a cube with a recurring decimal for bitshaving testing """
+    cube = set_up_test_cube()
+    # 1/9 fractions are recurring decimal and binary fractions
+    # good for checking number of digits remaining after bitshaving
+    cube.data = (
+        np.linspace(1.0 / 9.0, 1.0, 9, dtype=np.float32).reshape((1, 3, 3)) + 273
+    )
+    return cube
+
+
+@pytest.mark.parametrize("lsd", (0, 2, 5))
+@pytest.mark.parametrize("compress", (0, 2))
+def test_least_significant_digit(bitshaving_cube, tmp_path, lsd, compress):
+    """ Test the least significant digit for bitshaving output files"""
+    filepath = tmp_path / "temp.nc"
+    save_netcdf(
+        bitshaving_cube,
+        filepath,
+        compression_level=compress,
+        least_significant_digit=lsd,
+    )
+
+    # check that netcdf metadata has been set
+    data = Dataset(filepath, mode="r")
+    # pylint: disable=unsubscriptable-object
+    assert data.variables["air_temperature"].least_significant_digit == lsd
+
+    file_cube = load_cube(str(filepath))
+    abs_diff = np.abs(bitshaving_cube.data.data - file_cube.data.data)
+    # check that whole numbers are preserved
+    assert np.min(abs_diff) == 0.0
+    # check that modified data is accurate to the specified number of digits
+    assert np.median(abs_diff) < 10 ** (-1.0 * (lsd + 0.5))
+    assert np.mean(abs_diff) < 10 ** (-1.0 * (lsd + 0.5))
+    assert np.max(abs_diff) < 10 ** (-1.0 * lsd)
```

Comment:
Could we check that there are differences?

Reply:
Sure, I've added a check that the median and mean of abs_diff is above 0, I've had to reduce the highest lsd from 5 to 3 due to the number of d.p. when using float32